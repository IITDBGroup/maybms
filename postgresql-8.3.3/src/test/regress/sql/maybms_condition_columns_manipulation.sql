-- test for direct manipulation on condition columns 

create table s (a int, b int);
insert into s values (1,1), (1,2),(1,3),(2,1),(2,2),(2,3);

create table r as repair key a in s weight by b;

--insert condition columns
insert into r values (1,1,4,2,0.7);

--insert condition columns
insert into r values (1,1);
select * from r;

--modify condition columns
update r set _v0=4 where a = 1;

update r set _d0=4 where a = 1;

update r set _p0=4 where a = 1;

drop table r;
drop table s;


