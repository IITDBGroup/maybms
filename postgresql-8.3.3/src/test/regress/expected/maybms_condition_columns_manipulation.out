-- test for direct manipulation on condition columns 
create table s (a int, b int);
insert into s values (1,1), (1,2),(1,3),(2,1),(2,2),(2,3);
create table r as repair key a in s weight by b;
--insert condition columns
insert into r values (1,1,4,2,0.7);
ERROR:  INSERT has more expressions than target columns
--insert condition columns
insert into r values (1,1);
select * from r;
 a | b | _v0 | _d0 |   _p0    
---+---+-----+-----+----------
 1 | 1 |   1 |   1 | 0.166667
 1 | 2 |   1 |   3 | 0.333333
 1 | 3 |   1 |   2 |      0.5
 2 | 1 |   2 |   6 | 0.166667
 2 | 2 |   2 |   4 | 0.333333
 2 | 3 |   2 |   5 |      0.5
 1 | 1 |   3 |   7 |        1
(7 rows)

--modify condition columns
update r set _v0=4 where a = 1;
ERROR:  Query not supported: no direct manipulation allowed on condition columns.
update r set _d0=4 where a = 1;
ERROR:  Query not supported: no direct manipulation allowed on condition columns.
update r set _p0=4 where a = 1;
ERROR:  Query not supported: no direct manipulation allowed on condition columns.
drop table r;
drop table s;
